sigstore/verify/verifier.py

```diff
@@ -128,9 +128,18 @@ def _verify_signed_timestamp(
         for certificate_authority in cert_authorities:
             certificates = certificate_authority.certificates(allow_expired=True)
 
-            builder = VerifierBuilder()
-            for certificate in certificates:
-                builder.add_root_certificate(certificate)
+            # We expect at least a signing cert and a root cert but there may be intermediates
+            if len(certificates) < 2:
+                _logger.debug("Unable to verify Timestamp: cert chain is incomplete")
+                continue
+
+            builder = (
+                VerifierBuilder()
+                .tsa_certificate(certificates[0])
+                .add_root_certificate(certificates[-1])
+            )
+            for certificate in certificates[1:-1]:
+                builder = builder.add_intermediate_certificate(certificate)
```

Comment:
Flagging, nonblocking: we _could_ add these with `add_root_certificate` instead, since they're semantically root certs ("root" meaning "in the root program," not "is a self-signed certificate"). That would be _slightly_ faster, since our verification would then be one-hop (`TSR -> TSA -> TSA-CA`) rather than having to chain `TSA-CA` up to its "root."

In practice that's probably a marginal performance benefit, however.